xwiki-platform-core/xwiki-platform-model/xwiki-platform-model-validation/xwiki-platform-model-validation-default/src/main/java/org/xwiki/model/validation/internal/ReplaceCharacterEntityNameValidation.java

```diff
@@ -42,58 +42,82 @@
 @Singleton
 @Named(ReplaceCharacterEntityNameValidation.COMPONENT_NAME)
 public class ReplaceCharacterEntityNameValidation extends AbstractEntityNameValidation
-    implements Initializable
 {
+    /**
+     * An object encapsulating an entity name validation configuration.
+     */
+    private static class Configuration
+    {
+        private String[] forbiddenCharacters;
+
+        private String[] replacementCharacters;
+    }
+
     /**
      * Name of the component.
      */
     protected static final String COMPONENT_NAME = "ReplaceCharacterEntityNameValidation";
 
-    private String[] forbiddenCharacters;
-    private String[] replacementCharacters;
+    /**
+     * A map used as a cache of the current configurations of the wikis.
+     */
+    private final Map<String, Configuration> wikiConfigurationCache = new ConcurrentHashMap<>();
```

Comment:
Interesting, contrary than before you're allowing to have a different configuration for each wiki. It makes sense since right now I think we display the administration page in all subwikis, now I'm wondering if that's something we really want, or if we'd want to be able to configure it on the farm only: for example, in myxwiki.org you don't want the admin of a wiki to override the global farm entity name validation since it's based on some server restrictions in general.

Reply:
Interesting. In this case we might want to hide the configuration in subwikis (or at least make them edit the configuration located in the main wiki + relevant rights).
For now, after any wikis edit its configuration, it becomes the global one which is very confusing.

Reply:
> For now, after any wikis edit its configuration, it becomes the global one which is very confusing.

That's clearly a bug and it could make sense to report it, so we can discuss how we want to fix it there.

Reply:
https://jira.xwiki.org/browse/XWIKI-18670